File: 2018/01/solution.rb
#!/usr/bin/env ruby
require 'set'

# Read in and convert to integers
dat = File.readlines('input.txt').map { |s|
   Integer(s)
}

# Solve Part 1
sum = 0
dat.each { |i| sum += i }
puts "Part 1: #{sum}"

# Solve Part 2
freqs = Set[0]
sum = 0
stop = 0
while stop == 0
    dat.each { |i|
        sum += i
        if freqs.add?(sum) == nil
            stop = 1
            break
        end
    }
end

puts "Part 2: #{sum}"

File: 2018/01/solution.py
#!/usr/bin/env python3
"""Advent of Code 2019 - Day 1"""

import sys

def usage():
    """Prints usage string"""
    print("usage: solution [input-file]")

def main(argc, argv):
    """Main entry point"""
    if argc != 2:
        usage()
        exit(0)

    with open(argv[1], 'r') as fd:
        dat = [int(l) for l in fd.read().splitlines()]

    print('Part 1: {}'.format(sum(dat)))


    tmp = 0
    found = 0
    freqs = { 0 }
    while not found:
        for i in dat:
            tmp += i

            if tmp in freqs:
                found = 1
                break
            else:
                freqs.add(tmp)

    print('Part 2: {}'.format(tmp))

if __name__ == "__main__":
    main(len(sys.argv), sys.argv)
